### ISSUE TITLE
Vote: Renaming the project

### ISSUE THREAD

**Issue by carolynvs** [gomods/athens#664]
Suggestion: **Rename the project to something that most go developers would at least recognize as being a) related to go and b) solving a problem (such as dependency mgmt, proxies, modules, etc).**

I'm going to reply to this issue with suggestions that people can vote on using 👍 or 👎 . Feel free to contribute your own suggestions, but please it would really help if you formulated your replies in the same format, so that they are votable as well.

---
## Background

At GopherCon it became painfully obvious that continuing to use the codenames "Athens" and "Olympus" is hurting understanding of our project. No one understood what "Project Athens" was, that it was even associated with: go, modules, goproxy, etc. During our workshop, explaining the difference between Athens, proxy and Olympus was difficult and confusing for our new contributors. 

We chatted about this among the maintainers and during our dev meeting yesterday, and decided to vote on new names.

**Comment by carolynvs** [comment]
Project Name: Gomods
Sub-project Names:
* Gomods Proxy
* Gomods Registry
* Gomods (whatever else we may come up with)

**Comment by carolynvs** [comment]
Project Name: Gomods
Sub-project Names:
* Gopher Proxy
* Gopher Registry

**Comment by carolynvs** [comment]
Project Name: GopherPacks
Sub-project Names:
* GopherPacks Proxy
* GopherPacks Registry

**Comment by carolynvs** [comment]
ProjectName: GoMods
Sub-project Names:
* GoProxy
* GoRegistry

**Comment by marwan-at-work** [comment]
I think we should wait on renaming anything until we figure out what to do with Olympus on a technical and a architectural level. 

For example, if Olympus/Proxy ended up getting merged into *one thing*: then we can just call the server itself Athens and that's. In other words if someone asks "what's Athens?" We'll have a simple answer: it's a GOPROXY implementation. There's only one server inside the Athens repo, and that is the Athens server. 

But if Olympus becomes a big thing that's totally separate from the "proxy" that we currently have, then yes, it's worth looking at better naming of things.

**Comment by arschles** [comment]
good point @marwan-at-work. here's my proposal:

1. We make olympus a "registry" and call it the "gomods registry". you'd publish to the registry with a CLI - `gomods publish 1.3.1` or similar (and it could support custom domains maybe?)
2. we keep the proxy as it is now and call it the "gomods proxy". we could (and already do) run proxies publically too, so that people can `go get github.com/my/package`. packages still come out of VCSes

**Comment by marwan-at-work** [comment]
@arschles a much easier solution is to just have the registry be the only public one and wait on Go1.12 to fix this issue: https://github.com/golang/go/issues/26334

**Comment by marwan-at-work** [comment]
I'd love to suggest that our names should be unique in that it won't be confused with other Go Proxies that the open source community is building. For example: `gomods` and `goproxy` are easily confused and there's already a a [goproxy.io](https://github.com/goproxyio/goproxy) -- so I suggest that we shouldn't be `gomods`. 

For example, GopherPacks sounds the best so far. But other people might build GopherProxies or GopherMods and the community will easily confused them together.

I'll add suggestions as other comments once my creativity kicks in, if ever :)

**Comment by arschles** [comment]
At the [October 18, 2018 developer meeting](https://docs.google.com/document/d/1xpvgmR1Fq4iy1j975Tb4H_XjeXUQUOAvn0FximUzvIk/edit#heading=h.aor706plkyhh), we discussed #664. Since we decided in #772 that we're not gonna do a registry, we decided that the vote should change a little bit because we don't have to distinguish between the proxy and registry.

I said in the meeting that I would create a new vote between "Gomods proxy" and "Athens proxy." I thought about it and decided that we should move on and call this the Athens proxy since it's what we've been doing already.

I am going to close this issue now and finalize us on "Athens proxy" - let's move forward with our [v0.1.0](https://github.com/gomods/athens/milestone/2) beta release :smile:
